torch_tvm/compiler.cpp

```diff
@@ -222,29 +233,40 @@ void TVMCompiler::run(Stack& stack) {
     value_to_ivalue[value_input] = inputs[i];
   }
 
-  CompleteArgumentSpec spec{false, ArrayRef<IValue>(inputs)};
-
-  if (cache_.find(spec) == cache_.end()) {
+  if (!cache_ || (cache_ && (*cache_).invalid)) {
     for (auto& kv : value_to_ivalue) {
-      kv.first->inferTypeFrom(kv.second.toTensor());
+      // TODO: convince Fuser to NOT create TVMCompilationGroups
+      // if ANY of subgraph inputs weren't profiled
+      TORCH_INTERNAL_ASSERT(kv.first->type()->cast<ProfiledTensorType>());
     }
     // bail out mechanism: try to convert to Relay, if it fails to convert the
     // graph by any reason(i.e. op difference), depend on the user preference,
     // either throw or fall back to the JIT interpreter for execution
+    cache_ = TVMObject {};
     tvm::relay::Function tvm_func;
     try {
-      tvm_func = convertToRelay(subgraph_, ctx_, &cache_[spec].input_values);
+      tvm_func = convertToRelay(subgraph_, ctx_, &(*cache_).input_values);
+      // we compiled the subgraph successfully
+      (*cache_).invalid = false;
     } catch (const std::exception& e) {
+      (*cache_).invalid = true;
```

Comment:
When we failing back to JIT, this means some operators are not converted successfully due to operator semantic mismatch and other behaviors, this invalid flag will let the compiler to re-run the conversion everytime for the same inputs and it will always fail, so I am not sure if that flag would be necessary.

Reply:
exactly! we don't want to re-run if we already know it's going to fail! Re-running compilations might be pretty expensive